src/plugins/stake-plugins/generic/GenericStakePlugin.ts

```diff
@@ -0,0 +1,101 @@
+import { ChangeQuote, ChangeQuoteRequest, StakePlugin, StakePluginFactory, StakePolicy, StakePolicyFilter, StakePosition, StakePositionRequest } from '../types'
+import { makeGlifInfinityPoolAdapter } from './policyAdapters/GlifInfinityPoolAdapter'
+import { StakeAdapterConfig, StakePolicyAdapter } from './policyAdapters/types'
+import { StakePluginInfo, StakePolicyConfig } from './types'
+
+export const makeGenericStakePlugin =
```

Comment:
OK, so you are converting an "adapter" into a "plugin".

If writing a "plugin" is difficult because of all the boilerplate, then maybe we should simplify things so the "adapter" is all you really need to write, and *that* becomes the official plugin type.

I don't know how the existing plugins look. If there is an easy way to port them to this new "adapter" architecture, then we should just do it, and then just delete the whole stake plugin concept.

If we cannot port the existing stake plugins to "adapter", then why not? Are they more complicated? Do they need special logic? If there is something that makes the generic ones different, then maybe we should rename "Generic" to "EVM" or "Simple" or whatever that difference is.

Reply:
I originally attempted porting plugins, but found a lot of my time was going into this effort and not the actual task. I decided to design the new plugin with clean slate and start porting other plugins as a break away task to clean up tech debt. The reason is mainly a resource constraint; if I had more time, I would port the other plugins over to this adapter pattern and then flatten the abstractions.

Reply:
As long as there is an Asana task, this is fine.